src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/CertificateGeneratorRegistry.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.certificate.infra;
+
+import com.aws.greengrass.clientdevices.auth.certificate.CertificateGenerator;
+import lombok.Getter;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+/**
+ * Simple store for server certificate generators.
+ */
+public class CertificateGeneratorRegistry {
```

Comment:
The wrapper around the set I mentioned above, basically acts as a `Repository` - Terminology and what we store might change in the future